### DISCUSSION TITLE
Lock chunk does not exist because specified briefcase does not own any locks error

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#3085] [Q&A]
Hi , I am trying to insert an element in BriefcaseDb 
![image](https://user-images.githubusercontent.com/53996840/150678646-bb5e92c3-54a1-4853-84bd-c6142111d426.png)
but getting this error : 
![image](https://user-images.githubusercontent.com/53996840/150678680-736c9297-fc46-4afd-86fe-870fac7921cb.png)

The Policy is OptimisticPolicy, I tried to debug inside the request method , there I can see in the pendingrequest object, the codes array having 1 code value , but the locks array is empty

I believe that while inserting an element , the pendingrequest object should get filled with lock and code objects in their respective arrays

so, why this error is coming with lock(why the lock array is empty) ??

**Comment by Rahul-0108** [comment]
Any reason for this error ??

**Reply by kabentley** [reply]
You should note that the entire "concurrencyControl" system was removed/replaced in 3.0. If you're working on an editing application, you'll be a lot more productive there.

But, if you're using optimistic policy, locks are not used at all, so I don't understand what is wrong here. @swwilson-bsi ?
